include/cxx.h

```diff
@@ -14,6 +14,45 @@
 #include <basetsd.h>
 #endif
 
+#ifdef CXXBRIDGE_HAS_STRING_VIEW
+#error "string_view gets auto-detected"
+#endif
+
+// clang-format off
+// detect if string_view supported (C++17 required)
+#ifdef __has_include
+  #if __has_include(<version>)
+    // gcc >= 9, clang >= 9, msvc >= 19.22
+    #include <version>
+    #if __cpp_lib_string_view >= 201603L
+      #define CXXBRIDGE_HAS_STRING_VIEW
+    #endif
+  #else
+    // no <version> include
+    #if defined(__cpp_lib_string_view) && __cpp_lib_string_view >= 201603L
+      // gcc: 8, clang: 5 - 8, msvc: 19.15 - 19.21
+      #define CXXBRIDGE_HAS_STRING_VIEW
+    #else
+      // only include if compiled with c++17
+      #if (__GNUC__ >= 7 && __cplusplus >= 201703L) || (_MSVC_LANG >= 201703L)
+        // gcc: 7, msvc: 19.14
+        #define CXXBRIDGE_HAS_STRING_VIEW
+      #endif
+    #endif
+  #endif
+#else
+  // no __has_include
+  #if _MSC_VER >= 1910L && _MSVC_LANG > 201402L
+    // msvc: 19.10 requires c++latest (!)
```

Comment:
Older msvc compilers that supported string_view. Not sure what compiler versions you want to support, but I've included the ones I could get my hands on through godbolt.

There could be a discussion if it would make sense to support "c++latest" in cxx as a feature (for msvc).